controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -307,36 +310,33 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 	kcp.Status.UpdatedReplicas = int32(len(currentMachines))
 
 	replicas := int32(len(ownedMachines))
+
+	// set some basics in case the workload cluster is not yet available
 	kcp.Status.Replicas = replicas
-	readyMachines := int32(0)
+	kcp.Status.UnavailableReplicas = replicas
 
-	remoteClient, err := r.remoteClientGetter(ctx, r.Client, util.ObjectKey(cluster), r.scheme)
+	workloadCluster, err := r.managementCluster.GetWorkloadCluster(ctx, util.ObjectKey(cluster))
+	// Not being able to get the workload cluster is fine, but if this message persists there is a problem
 	if err != nil {
-		if cause := errors.Cause(err); !apierrors.IsNotFound(cause) && !apierrors.IsTimeout(cause) {
-			return errors.Wrap(err, "failed to create remote cluster client")
-		}
-	} else {
-		for i := range ownedMachines {
-			node, err := getMachineNode(ctx, remoteClient, ownedMachines[i])
-			if err != nil {
-				return errors.Wrap(err, "failed to get referenced Node")
-			}
-			if node == nil {
-				continue
-			}
-			if node.Spec.ProviderID != "" {
-				readyMachines++
-			}
-		}
+		r.Log.Info("unable to get workload cluster", "err", fmt.Sprintf("%s", err))
+		return nil
```

Comment:
You are right that this change set would modify behavior in a subtle way. I think this should handle the error cases exactly as they were being handled before and so not change any behavior of this function.

Do you think a `capierrors.RequeueAfterError` is more appropriate than returning the errors.Wrap?

Reply:
I think my worry with not using RequeueAfterError here is that we could hit the 10 retries during the normal error path prior to the remote APIServer being available, which would lead to needing to wait for the resync interval to continue updating the status (or really any other operations after the creation of the initial control plane instance), while waiting for that initial machine to finish booting/initializing.

Reply:
The way the defer is structured there is no way to send up a requeue after error from updateStatus. If you think this should definitely use requeueAfter then we'll need a new issue / PR to refactor how the defer works